speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/SocketBlackBox.inl

```diff
@@ -183,14 +142,18 @@ void SocketBlackBox<DriveSituation>::DeserializeBlackBoxResults(const char* p_da
 
 /// @brief Sends serialized drive situation to black box. Deserializes data received by black box.
 ///        Makes decisions from received data.
-/// @param p_driveSituation Drive situation to base decisions off.
+/// @param p_car       Car to base decisions off.
+/// @param p_situation Situation to base decisions off.
+/// @param p_tickCount The current tick count.
 /// @return returns decision array.
-template<class DriveSituation>
-bool SocketBlackBox<DriveSituation>::GetDecisions(DriveSituation& p_driveSituation, DecisionTuple& p_decisions)
+template <class BlackBoxData>
+bool SocketBlackBox<BlackBoxData>::GetDecisions(tCarElt* p_car, tSituation* p_situation, int p_tickCount, DecisionTuple& p_decisions)
 {
     if (!m_server.GetData(m_buffer, SBB_BUFFER_SIZE)) return false;
     msgpack::sbuffer sbuffer;
-    SerializeDriveSituation(sbuffer, p_driveSituation);
+    delete m_currentData;
+    m_currentData = new BlackBoxData(p_car, p_situation, p_tickCount);
```

Comment:
Why not give BlackBoxData as parameter in the first place, if the only thing you do here is construct a BlackBoxData object from the given parameters?

Reply:
This data is created on the heap and needs to stay there until the black box has finished working with it. If I keep making data every step that would mean deleting the pointer as well, and then the black box doesn't know where it should find the data